src/main/java/com/checkout/four/CheckoutApiImpl.java

```diff
@@ -140,4 +141,10 @@ public ApiClient getTransfersClient(final CheckoutConfiguration configuration) {
                 .orElse(null);
     }
 
+    public ApiClient getBalancesClient(final CheckoutConfiguration configuration) {
+        return Optional.ofNullable(configuration.getEnvironment())
+                .map(environment -> new ApiClientImpl(configuration, environment.getBalancesApiURI()))
+                .orElse(null);
```

Comment:
So, `configuration.getEnvironment()` cannot be null right? It shouldn't be. We cannot allow a null environment at this point, this nullifies the need of an Optional. I would just keep things simple here:

```
return new ApiClientImpl(configuration, environment.getBalancesApiURI())
```